### PR TITLE
[Feat] Add rescale functional and transformation

### PR DESCRIPTION
**Pull request by pmeier**
### Description

The `kornia.Rescale` transformation and the accompanying `kornia.rescale` functional change the size of the image similar to `kornia.Resize` and `kornia.resize`. The difference is the intention when to use them: While one would use `resize` when he wants a specific size of the new image, `rescale` should be used if the new size is best given relative to the original. That means, it is easier to specifiy things like half or double the original size with `rescale`.

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [x] Did you discuss the functionality or any breaking changes before ?
- [x] **Pass all tests**: did you test in local ? `make test`
- [x] Unittests: did you add tests for your new functionality ?
- [x] Documentations: did you build documentation ? `make build-docs`
- [x] Implementation: is your code well commented and follow conventions ? `make lint`
- [x] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [x] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

**Comment by edgarriba** [comment]
@pmeier could you send an extra PR improving documentation, like specifying i/o tensor shape and including a small snippet example ? Linking functions seems to be broken.

Besides, what do you think about having `Zoom` functions as well ?

**Comment by pmeier** [comment]
@edgarriba 

> could you send an extra PR improving documentation, like specifying i/o tensor shape and including a small snippet example ?

I've based the docstring on `kornia.Resize` which also does not include this:

https://github.com/kornia/kornia/blob/25531b6a77b494378d8f6e3818c29d9119763abc/kornia/geometry/transform/affwarp.py#L381-L398

Where do you want me to put this information?

> Linking functions seems to be broken.

What do you mean by this? The only intersphinx link to `torch.nn.functional` works for me.

> Besides, what do you think about having `Zoom` functions as well ?

What would that transformation do?

- `Resize` / `Rescale` change the image dimensions
- `Scale` enlarges the motif without changing the dimensions

Is there anything left that `Zoom` should do?

**Comment by edgarriba** [comment]
@pmeier I think that for proper linking in docs you need to include the full package sequence - kornia.geometry.transfroms.Rescale. Another thing to think about, it's the way we maintain duplicated documentation for functional/modules.

Yeah, maybe you are right, Scale already does the thing. Maybe we just need to improve the documentation on those functionalities, explaining the difference between than and add small snippet examples, i/o shapes, etc.
